echopype/convert/set_groups_ad2cp.py

```diff
@@ -252,79 +261,22 @@ def set_beam(self) -> List[xr.Dataset]:
-                            "coordinate_system": "coordinate_system",
-                            "num_cells": "number_of_cells",
-                            "blanking": "blanking",
-                            "cell_size": "cell_size",
-                            "velocity_range": "velocity_range",
-                            "echosounder_frequency": "echosounder_frequency",
-                            "ambiguity_velocity": "ambiguity_velocity",
-                            "dataset_description": "data_set_description",
-                            "transmit_energy": "transmit_energy",
-                            "velocity_scaling": "velocity_scaling",
-                            "velocity_data_burst": "velocity",
-                            "amplitude_data_burst": "amplitude",
-                            "correlation_data_burst": "correlation",
-                        }
-                    )
-                )
-
-                self._beamgroups.append(
-                    {
-                        "name": f"Beam_group{len(self._beamgroups) + 1}",
-                        "descr": (
-                            "contains echo intensity, velocity and correlation data "
-                            "as well as other configuration parameters from the Burst mode."
-                        ),
-                    }
-                )
-                break
-        # average
         for packet in self.parser_obj.packets:
             if packet.is_average():
-                beam_groups.append(
-                    self._make_dataset(
-                        {
-                            "num_beams": "number_of_beams",
-                            "coordinate_system": "coordinate_system",
-                            "num_cells": "number_of_cells",
-                            "blanking": "blanking",
-                            "cell_size": "cell_size",
-                            "velocity_range": "velocity_range",
-                            "echosounder_frequency": "echosounder_frequency",
-                            "ambiguity_velocity": "ambiguity_velocity",
-                            "dataset_description": "data_set_description",
-                            "transmit_energy": "transmit_energy",
-                            "velocity_scaling": "velocity_scaling",
-                            "velocity_data_average": "velocity",
-                            "amplitude_data_average": "amplitude",
-                            "correlation_data_average": "correlation",
-                        }
-                    )
-                )
+                beam_groups_exist.add(BeamGroup.AVERAGE)
+            elif packet.is_burst():
+                beam_groups_exist.add(BeamGroup.BURST)
+            elif packet.is_echosounder():
+                beam_groups_exist.add(BeamGroup.ECHOSOUNDER)
+            elif packet.is_echosounder_raw():
+                beam_groups_exist.add(BeamGroup.ECHOSOUNDER_RAW)
```

Comment:
Seems like you can break the look if all 4 possible groups already exist in the `beam_groups_exist` set -- though how much this helps on the speed depends on the actual number of packets in the file.